pkg/values.go

```diff
@@ -0,0 +1,585 @@
+package pkg
+
+import (
+	"bytes"
+	"errors"
+	"text/template"
+
+	apierrors "k8s.io/kubernetes/pkg/api/errors"
+	kcl "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+const (
+	offCluster = "off-cluster"
+	onCluster  = "on-cluster"
+)
+
+type valuesConfig struct {
+	StorageLocation       string
+	DatabaseLocation      string
+	RedisLocation         string
+	InfluxDBLocation      string
+	GrafanaLocation       string
+	RegistryLocation      string
+	RegistryHostPort      string
+	ImagePullSecretPrefix string
+	S3                    s3
+	GCS                   gcs
+	Azure                 azure
+	Swift                 swift
+	Postgres              postgres
+	Controller            controller
+	Redis                 redis
+	Grafana               grafana
+	InfluxDB              influxDB
+	ECR                   ecr
+	GCR                   gcr
+	OffClusterRegistry    offClusterRegistry
+	Router                router
+}
+
+type s3 struct {
+	AccessKey      string
+	SecretKey      string
+	Region         string
+	RegistryBucket string
+	DatabaseBucket string
+	BuilderBucket  string
+}
+
+type gcs struct {
+	KeyJSON        string
+	RegistryBucket string
+	DatabaseBucket string
+	BuilderBucket  string
+}
+
+type azure struct {
+	AccountName       string
+	AccountKey        string
+	RegistryContainer string
+	DatabaseContainer string
+	BuilderContainer  string
+}
+
+type swift struct {
+	UserName          string
+	Password          string
+	Tenant            string
+	AuthURL           string
+	AuthVersion       string
+	RegistryContainer string
+	DatabaseContainer string
+	BuilderContainer  string
+}
+
+type controller struct {
+	AppPullPolicy    string
+	RegistrationMode string
+}
+
+type postgres struct {
+	Name     string
+	UserName string
+	Password string
+	Host     string
+	Port     string
+}
+
+type redis struct {
+	DB       string
+	Host     string
+	Port     string
+	Password string
+}
+
+type grafana struct {
+	User     string
+	Password string
+}
+
+type influxDB struct {
+	Database string
+	URL      string
+	User     string
+	Password string
+}
+
+type ecr struct {
+	AccessKey  string
+	SecretKey  string
+	Region     string
+	RegistryID string
+	HostName   string
+}
+
+type gcr struct {
+	KeyJSON  string
+	HostName string
+}
+
+type offClusterRegistry struct {
+	HostName     string
+	Organization string
+	UserName     string
+	Password     string
+}
+
+type router struct {
+	DHParam string
+}
+
+const (
+	valuesTemplate = `#
```

Comment:
file would mean i have to read it again and convert to string to use in the template.Parse() function